src/device.cpp

```diff
@@ -845,9 +845,7 @@ bool cvk_device::supports_capability(spv::Capability capability) const {
     case spv::CapabilityVulkanMemoryModel:
         return m_features_vulkan_memory_model.vulkanMemoryModel;
     case spv::CapabilityShaderNonUniform:
-        return m_properties.apiVersion >= VK_MAKE_VERSION(1, 2, 0) ||
-               is_vulkan_extension_enabled(
-                   VK_EXT_DESCRIPTOR_INDEXING_EXTENSION_NAME);
+        return supports_non_uniform_pointers();
```

Comment:
They're not really non-uniform "pointers". Maybe `supports_non_uniform_decoration`?

Reply:
Done